mesh/mesh.hpp

```diff
@@ -2433,7 +2440,199 @@ class Mesh
 
 /** Overload operator<< for std::ostream and Mesh; valid also for the derived
     class ParMesh */
-std::ostream &operator<<(std::ostream &out, const Mesh &mesh);
+std::ostream &operator<<(std::ostream &os, const Mesh &mesh);
+
+/// @brief Print function for Mesh::FaceInformation.
+std::ostream& operator<<(std::ostream &os, const Mesh::FaceInformation& info);
+
+
+// Class containing a minimal description of a part (a subset of the elements)
+// of a Mesh and its connectivity to other parts. The main purpose of this class
+// is to be communicated between MPI ranks for repartitioning purposes. It can
+// also be used to implement parallel mesh I/O functions with partitionings that
+// have number of parts different from the number of MPI tasks.
+//
+// Note: parts of NURBS or non-conforming meshes cannot be fully described by
+// this class alone.
+class MeshPart
+{
+protected:
+   struct Entity { int geom; int num_verts; const int *verts; };
+   struct EntityHelper
+   {
+      int dim, num_entities;
+      int geom_offsets[Geometry::NumGeom+1];
+      typedef const Array<int> entity_to_vertex_type[Geometry::NumGeom];
+      entity_to_vertex_type &entity_to_vertex;
+
+      EntityHelper(int dim_,
+                   const Array<int> (&entity_to_vertex_)[Geometry::NumGeom]);
+      Entity FindEntity(int bytype_entity_id);
+   };
+
+public:
+   // Reference space dimension of the elements
+   int dimension;
```

Comment:
Should these data members be protected?

Reply:
The structure `MeshPart` is primarily meant to be just a container. I wanted to keep its data members accessible so it can be constructed by `MeshPartitioner` and also leave the door open for users to construct or manipulate it if they want to.